model_compiler_2/src/model_compiler/openvino_util.py

```diff
@@ -1,10 +1,153 @@
+
+    def _find_output_info(self, layer):
+        for key in layer.ports:
+            for edge in self.edges:
+                if edge.to_layer == layer.id and edge.to_port == key:
+                    output_layer = self._find_layer_by_id(edge.from_layer)
+                    return (output_layer.name,) + output_layer.ports[edge.from_port]
+        raise ValueError('can not find layer by the tag "Result" layer')
+
+    def _find_layer_by_id(self, layer_id):
+        return next(layer for layer in self.layers if layer.id == layer_id)
+
+    def _get_outputs_info(self):
+        return [self._find_output_info(layer) for layer in self.layers if layer.type == 'Result']
+
+    def get_inputs(self):
+        inputs = []
+        for name, shape, data_type in self._get_inputs_info():
+            inputs.append(ModelInput(name=name,
+                                     data_type=DataType.from_openvino_data_type(data_type).to_tf_data_type(),
+                                     format=ModelInput.FORMAT_NONE,  # pylint: disable=no-member
+                                     dims=[-1 if dim is None else dim for dim in shape[1:]]))
+        return inputs
+
+    def get_outputs(self):
+        outputs = []
+        for name, shape, data_type in self._get_outputs_info():
+            outputs.append(ModelOutput(name=name,
+                                       data_type=DataType.from_openvino_data_type(data_type).to_tf_data_type(),
+                                       dims=[-1 if dim is None else dim for dim in shape[1:]]))
+        return outputs
+
+
+class Config(NamedTuple):
+    input_names: Optional[str]
+    input_shapes: Optional[str]
+    output_names: Optional[str]
+    max_batch_size: Optional[int]
+    enable_nhwc_to_nchw: Optional[bool]
+
+    @staticmethod
+    def from_json(value: Mapping[str, Any]) -> 'Config':
+        input_names = value.get('input_names')
+        input_shapes = value.get('input_shapes')
+        output_names = value.get('output_names')
+        max_batch_size = value.get('max_batch_size')
+        enable_nhwc_to_nchw = value.get('enable_nhwc_to_nchw')
+        return Config(input_names=input_names,
+                      input_shapes=input_shapes,
+                      output_names=output_names,
+                      max_batch_size=max_batch_size,
+                      enable_nhwc_to_nchw=enable_nhwc_to_nchw)
+
+    @staticmethod
+    def from_env(env: Mapping[str, Any]) -> 'Config':
+        input_names = env.get('INPUT_NAMES')
+        input_shapes = env.get('INPUT_SHAPES')
+        output_names = env.get('OUTPUT_NAMES')
+        max_batch_size = env.get('MAX_BATCH_SIZE')
+        enable_nhwc_to_nchw = env.get('ENABLE_NHWC_TO_NCHW')
+        if max_batch_size is not None:
+            max_batch_size = int(max_batch_size)
+        if enable_nhwc_to_nchw is not None:
+            enable_nhwc_to_nchw = ast.literal_eval(enable_nhwc_to_nchw)
```

Comment:
Conventionally, use `'0'` and `'1'` to represent boolean values in environment variables.